### ISSUE TITLE
Incorrect url rewrite record pagination

### ISSUE THREAD

**Issue by therool** [magento/data-migration-tool#682]
### Preconditions

1. Magento 1.9.0.1 to Magento 2.3.1
2. This seems to affect all versions of magento as the issue is with the [Version191to2000.php](https://github.com/magento/data-migration-tool/blob/2.3.1/src/Migration/Step/UrlRewrite/Version191to2000.php), [Version11300to2000.php](https://github.com/magento/data-migration-tool/blob/2.3.1/src/Migration/Step/UrlRewrite/Version11300to2000.php), [Version11410to2000.php](https://github.com/magento/data-migration-tool/blob/2.3.1/src/Migration/Step/UrlRewrite/Version11410to2000.php)

### Steps to reproduce

1. Have at least more than 50000 url rewrites
2. Start the migration process
3. Will receive ```[PDOException (23000)]                                                                           
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '50852' for key 'PRIMARY'```

### Expected result

1. There should be no primary key constraint violations at url rewrite step as each record should be processed only once during url rewrite step

### Actual result

1. Will receive ```[PDOException (23000)]                                                                           
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '50852' for key 'PRIMARY'```
![image](https://user-images.githubusercontent.com/15800061/56355904-2605e580-61e0-11e9-9e97-d7461b85fe93.png)

### Additional notes
The issue is caused that the ```$this->source->setLastLoadedRecord``` method is not being called and the data source pagination second page is not comply with the first result page query conditions.

Pagination queries:
1. page 0
```sql
SELECT `core_url_rewrite`.* FROM `core_url_rewrite` WHERE (`url_rewrite_id` >= 0) ORDER BY `url_rewrite_id` ASC
```
2. page 1 - Notice that the order condition is lost and the records do not comply with the query conditions above
```sql
SELECT `core_url_rewrite`.* FROM `core_url_rewrite` LIMIT 50000 OFFSET 50000
```
Exception trace:
```bash
In Mysql.php line 183:
                                                                                                   
  [PDOException (23000)]                                                                           
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '50852' for key 'PRIMARY'  
                                                                                                   

Exception trace:
 () at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/ResourceModel/Adapter/Mysql.php:183
 PDOStatement->execute() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/ResourceModel/Adapter/Mysql.php:183
 Migration\ResourceModel\Adapter\Mysql->insertMultiple() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/ResourceModel/Adapter/Mysql.php:140
 Migration\ResourceModel\Adapter\Mysql->insertRecords() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/ResourceModel/Destination.php:46
 Migration\ResourceModel\Destination->saveRecords() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/Step/UrlRewrite/Version191to2000.php:225
 Migration\Step\UrlRewrite\Version191to2000->data() at n/a:n/a
 call_user_func() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/Step/UrlRewrite/Version191to2000.php:242
 Migration\Step\UrlRewrite\Version191to2000->perform() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/Mode/AbstractMode.php:81
 Migration\Mode\AbstractMode->runStage() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/Mode/Data.php:124
 Migration\Mode\Data->runData() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/Mode/Data.php:69
 Migration\Mode\Data->run() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/data-migration-tool/src/Migration/Console/MigrateDataCommand.php:54
 Migration\Console\MigrateDataCommand->execute() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/symfony/console/Application.php:893
 Symfony\Component\Console\Application->doRunCommand() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/symfony/console/Application.php:262
 Symfony\Component\Console\Application->doRun() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/magento/framework/Console/Cli.php:102
 Magento\Framework\Console\Cli->doRun() at /var/www/monobunt/magento2/projects/ammerer/htdocs/vendor/symfony/console/Application.php:145
 Symfony\Component\Console\Application->run() at /var/www/monobunt/magento2/projects/ammerer/htdocs/bin/magento:23
```

**Comment by victor-v-rad** [comment]
Hi @therool 

Thank you for reporting this issue. Internal ticket MAGETWO-99323 to fix it

**Comment by therool** [comment]
@victor-v-rad hello, I did create a pull request https://github.com/magento/data-migration-tool/pull/683 which included the necessary fix but I see that you already pushed the fixes to the 2.3-develop branch so you can close the pull request https://github.com/magento/data-migration-tool/pull/683

**Comment by victor-v-rad** [comment]
Oh, sorry! Pull requests from community are so rare and I did not notice it.  I reverted my commit and merged your. Thank you @therool
